net/socks5/socks5.go

```diff
@@ -199,19 +200,30 @@ func (c *Conn) handleRequest() error {
 	errc := make(chan error, 2)
 	go func() {
 		_, err := io.Copy(c.clientConn, srv)
-		if err != nil {
+		if errors.Is(err, io.EOF) {
```

Comment:
https://golang.org/pkg/io/#Copy says:

> Copy copies from src to dst until either EOF is reached on src or an error occurs. It returns the number of bytes copied and the first error encountered while copying, if any.
>
>A successful Copy returns err == nil, not err == EOF. Because Copy is defined to read from src until EOF, it does not treat an EOF from Read as an error to be reported.

So where are the io.EOFs coming from?

There's no linked bug or new test, so not sure the context/repro here?

Reply:
In working with Cloud Run, it was getting frequent disconnects followed by immediate reconnects. Implementing this seemed to improve things so it no longer disconnected. However I couldn't make the problem happen in a unit test.

Later I discovered a different mistake I'd made in the Cloud Run app, use of math/rand instead of crypto/rand meant that it wasn't generating an actually unique ID per instance. I think that was the root cause of the disconnects (the server kicks off the older client using the same ID). This change appeared to help because of happenstance, of only one instance running while trying it.